core/controllers/dashboard.py

```diff
@@ -152,44 +153,21 @@ def _get_intro_card_color(category):
                 subscription_services.get_collection_ids_subscribed_to(
                     self.user_id)))
 
-        explorations_list = []
+        explorations_list = summary_services.get_displayable_exp_summary_dicts(
+            subscribed_exploration_summaries)
         collections_list = []
 
-        for exp_summary in subscribed_exploration_summaries:
-            if exp_summary is None:
-                continue
-
-            feedback_thread_analytics = feedback_services.get_thread_analytics(
-                exp_summary.id)
-            # TODO(sll): Reuse _get_displayable_exp_summary_dicts() in
-            # summary_services, instead of replicating it like this.
-            explorations_list.append({
-                'id': exp_summary.id,
-                'title': exp_summary.title,
-                'category': exp_summary.category,
-                'objective': exp_summary.objective,
-                'language_code': exp_summary.language_code,
-                'last_updated': utils.get_time_in_millisecs(
-                    exp_summary.exploration_model_last_updated),
-                'created_on': utils.get_time_in_millisecs(
-                    exp_summary.exploration_model_created_on),
-                'status': exp_summary.status,
-                'community_owned': exp_summary.community_owned,
-                'thumbnail_icon_url': (
-                    utils.get_thumbnail_icon_url_for_category(
-                        exp_summary.category)),
-                'thumbnail_bg_color': utils.get_hex_color_for_category(
-                    exp_summary.category),
-                'ratings': exp_summary.ratings,
-                'num_open_threads': (
-                    feedback_thread_analytics.num_open_threads),
-                'num_total_threads': (
-                    feedback_thread_analytics.num_total_threads),
-            })
+        feedback_thread_analytics = (
+            feedback_services.get_thread_analytics_multi(
+                subscription_services.get_exploration_ids_subscribed_to(
```

Comment:
Why do the subscription_services.get_exploration_ids_subscribed_to() call again? Compute the result of that once, and store it in a local variable here -- this saves a datastore call.